addon/types/map_export/brush.py

```diff
@@ -40,7 +40,21 @@ def sort_into_parts(bm: bmesh.types.BMesh):
     return parts
 
 
-def calc_uv_axes(settings: typing.Any, bm: bmesh.types.BMesh, face: bmesh.types.BMFace):
+def get_texture_size(obj: bpy.types.Object, face: bmesh.types.BMFace):
+    texture_side_length = -1
+
+    if face.material_index < len(obj.material_slots):
```

Comment:
Might want to use `face.material_index in range(len(obj.material_slots))`  to make sure that negative numbers don't cause any trouble; though I'm not sure if those can happen so it probably doesn't matter.